[9.0][purchase_request_to_procurement] Allow to select all products, even when they are not purchaseable.

JordiBForgeFlow opened the pull request:
Fixes #427

JordiBForgeFlow committed:
Allow to select all products, even when they are not purchaseable.

JordiBForgeFlow commented:
cc @lreficent @benwillig

LoisRForgeFlow commented:
I would take the ocassion to also attend [this comment](https://github.com/OCA/purchase-workflow/pull/424#issuecomment-312199138).

JordiBForgeFlow committed:
Fixes two things:
- When you copy a purchase request the cancelled boolean should be reset.
- Users should not be able to duplicate just purchase request lines

LoisRForgeFlow reviewed (approved):
LGTM :+1:

MiquelRForgeFlow reviewed (approved):
nice :+1:

benwillig commented on purchase_request/models/purchase_request.py:
```diff
@@ -255,7 +255,7 @@ def _compute_supplier_id(self):
                                      readonly=True, copy=False)
     cancelled = fields.Boolean(
-        string="Cancelled", readonly=True, default=False)
+        string="Cancelled", readonly=True, default=False, copy=False)
```
Why removing ```copy=False``` from the field declaration?
